Source/CBLCookieStorage.m

```diff
@@ -0,0 +1,424 @@
+//
+//  CBLCookieStorage.m
+//  CouchbaseLite
+//
+//  Created by Pasin Suriyentrakorn on 2/18/15.
+//  Copyright (c) 2015 Couchbase, Inc. All rights reserved.
+//
+
+#import "CBLCookieStorage.h"
+#import "CBLDatabase.h"
+#import "CBLDatabase+Internal.h"
+#import "CBLMisc.h"
+#import "Logging.h"
+
+
+NSString* const CBLCookieStorageCookiesChangedNotification = @"CookieStorageCookiesChanged";
+NSString* const CBLCookieStorageAcceptPolicyChangedNotification = @"CookieStorageAcceptPolicyChanged";
+
+#define kLocalDocKeyPrefix @"cbl_cookie_storage"
+#define kLocalDocCookiesKey @"cookies"
+
+@interface CBLCookieStorage ()
+- (NSString*) localDocKey;
+- (void) loadCookies;
+- (BOOL) deleteCookie: (NSHTTPCookie*)aCookie outIndex: (NSUInteger*)outIndex;
+- (BOOL) saveCookies: (NSError **)error;
+- (void) pruneExpiredCookies;
+- (BOOL) isExpiredCookie: (NSHTTPCookie*)cookie;
+- (BOOL) isDomainMatchedBetweenCookie: (NSHTTPCookie*)cookie andUrl: (NSURL*)url;
+- (BOOL) isPathMatchedBetweenCookie: (NSHTTPCookie*)cookie andUrl: (NSURL*)url;
+- (void) notifyCookiesChanged;
+@end
+
+
+@implementation CBLCookieStorage
+{
+    NSMutableArray* _cookies;
+    CBLDatabase* _db;
+    NSString* _storageKey;
+}
+
+
+@synthesize cookieAcceptPolicy = _cookieAcceptPolicy;
+
+
+- (instancetype) initWithDB: (CBLDatabase*)db storageKey: (NSString*)storageKey {
+    self = [super init];
+    if (self) {
+        Assert(db != nil, @"database cannot be nil.");
+        Assert(storageKey != nil, @"storageKey cannot be nil.");
+
+        _db = db;
+        _storageKey = storageKey;
+        self.cookieAcceptPolicy = NSHTTPCookieAcceptPolicyAlways;
+
+        [self loadCookies];
+    }
+    return self;
+}
+
+
+- (void) setCookieAcceptPolicy: (NSHTTPCookieAcceptPolicy)cookieAcceptPolicy {
+    @synchronized(self) {
+        if (cookieAcceptPolicy == NSHTTPCookieAcceptPolicyOnlyFromMainDocumentDomain)
+            Warn(@"%@: Currently NSHTTPCookieAcceptPolicyOnlyFromMainDocumentDomain \
+                 is not supported.", self);
+
+        if (_cookieAcceptPolicy != cookieAcceptPolicy) {
+            _cookieAcceptPolicy = cookieAcceptPolicy;
+            [[NSNotificationCenter defaultCenter] postNotificationName: CBLCookieStorageAcceptPolicyChangedNotification
```

Comment:
I removed the CBLCookieStorageAcceptPolicyChangedNotification notification but I still wanna keep the warning though.